W-14221636: Update Feature Flagging doc (#2562)
mulesoft/docs-mule-runtime: modules/ROOT/pages/feature-flagging.adoc

*Issue ID*

* W-11529823
<.^|`com.mulesoft.dw.xml_reader.honourMixedContentStructure`
|When this property is set to true, DataWeave retains a mixed-content structure instead of grouping text with mixed content into a single text field.

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-11071481
<.^|`ENFORCE_ERROR_TYPES_VALIDATION` not configurable by system property

> |When enabled, error types validations will be enforced, even for error handlers/components that are not being referenced.

Reviewer: ```suggestion
<.^|`ENFORCE_ERROR_TYPES_VALIDATION` 
|When enabled, error types validations are enforced, even for error handlers/components that are not being referenced.
This feature isn't configurable by system property.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-10619787
<.^|`disable.attribute.parameter.whitespace.trimming`

> |When enabled, the Runtime will trim whitespaces from parameter values defined at the attribute level in the dsl.

Reviewer: ```suggestion
|When enabled, Mule trims whitespaces from parameter values defined at the attribute level in the DSL.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* MULE-19803
<.^|`disable.pojo.text.parameter.whitespace.trimming`

> |When enabled, the Runtime will trim whitespaces from CDATA text parameter of pojos in the dsl.

Reviewer: ```suggestion
|When enabled, Mule trims whitespaces from CDATA text parameter of pojos in the DSL.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* MULE-20048
<.^|`enforce.expression.validation`

> |When enabled, expression validations will be enforced for `targetValue`, not allowing a literal value.

Reviewer: ```suggestion
|When enabled, expression validations are enforced for `targetValue`, not allowing a literal value.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* MULE-19987
<.^|`enforce.dw.expression.validation`

> |When enabled, expression validations will be enforced for all DataWeave expressions.

Reviewer: ```suggestion
|When enabled, expression validations are enforced for all DataWeave expressions.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* MULE-19967
<.^|`force.runtime.profiling.consumers.enablement`

> |When enabled, profiling consumers implemented by the runtime will be enabled by default.

Reviewer: ```suggestion
|When enabled, profiling consumers implemented by Mule are enabled by default.
```

*Available Since*

* 4.5.0

* 4.4.0-202202

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* MULE-19967
<.^|`mule.disable.registryBootstrapOptionalEntries`

> |When enabled, `optional` attribute in entries in a `registry-bootstrap.properties` will be ignored.

Reviewer: ```suggestion
|When enabled, `optional` attribute in entries in a `registry-bootstrap.properties` are ignored.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-10736301
<.^|`mule.disable.applyObjectProcessor`

> |When enabled, `org.mule.runtime.core.privileged.registry.ObjectProcessor` implementations will not be applied on objects registered into the `SimpleRegistry`.

Reviewer: ```suggestion
|When enabled, `org.mule.runtime.core.privileged.registry.ObjectProcessor` implementations aren't applied on objects registered into the `SimpleRegistry`.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* MULE-11737
<.^|`mule.deployment.validateAppModelWithRegionClassloader`

> |When enabled, the application model will be validated with the region classloader. When disabled, it will be validated with the application classloader.

Reviewer: ```suggestion
|When enabled, the application model is validated with the region classloader. When disabled, it is validated with the application classloader.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-10808757
<.^|`mule.suppress.mule.exceptions`

> |When enabled, error suppression will happen. This will affect, for example, the Web Service Consumer connector and the Until Successful scope that will always report errors from their corresponding namespaces (MULE and WSC). Suppressed errors will be treated as underlying causes.

Reviewer: ```suggestion
|When enabled, error suppression happens. This affects, for example, the Web Service Consumer connector and the Until Successful scope that always reports errors from their corresponding namespaces (MULE and WSC). Suppressed errors are treated as underlying causes.
```

*Available Since*

* 4.5.0
* 4.4.0-202210

*Enabled by Default Since*

* 4.5.0
* 4.4.0-202210
* 4.3.0-202210

*Issue ID*

* W-11308645
<.^|`mule.foreachRouterRejectsMapExpressions`

> |When enabled, the 'foreach' router will generate an IllegalArgumentException if the collection expression evaluates to a java.util.Map.

Reviewer: ```suggestion
|When enabled, the 'foreach' router generates an `IllegalArgumentException` if the collection expression evaluates to a `java.util.Map`.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-12207110
<.^|`mule.honour.insecure.tls.configuration`

> |When enabled, the insecure TLS configuration will be honoured even if there are fields of the TrustStore configured.

Reviewer: ```suggestion
|When enabled, the insecure TLS configuration is honored even if there are fields of the TrustStore configured.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-10822938
<.^|`mule.transaction.sink.index`

> |When enabled, flux sinks will be cached using index as part of the key. If a sink is already in use, new sink will be created.

Reviewer: ```suggestion
|When enabled, flux sinks are cached using index as part of the key. If a sink is already in use, a new sink is created.
```

*Available Since*

* 4.4.0

*Enabled by Default Since*

* 4.0.0

* 4.1.0

* 4.2.0

* 4.3.0

* 4.4.0

* 4.5.0

*Issue ID*

* W-12128703
<.^|`mule.put.trace.id.and.span.id.in.mdc`

> |When enabled, the trace id and span id will be added to the MDC when available.

Reviewer: ```suggestion
|When enabled, the trace ID and span ID are added to the MDC when available.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-12979787
<.^|`mule.enable.mule.specific.tracing.information`

> |When enabled the `ancestor-mule-span-id` value will be added in the trace state when a span is propagated.

Reviewer: ```suggestion
|When enabled, the `ancestor-mule-span-id` value is added to the trace state when a span is propagated.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* Not enabled by default in any Mule version

*Issue ID*

* W-13215870
<.^|`mule.enable.policy.context.parallel.scopes`

> |When enabled, a new (Source) Policy Context is created for the execution of parallel scopes: ParallelForeach, ScatterGather and Async.

Reviewer: ```suggestion
|When enabled, a new (Source) Policy Context is created for the execution of parallel scopes: ParallelForeach, ScatterGather, and Async.
```

*Available Since*

* 4.5.0

* 4.4.0-202306

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-13509911
<.^|`mule.disableJmx.for.commons.pool2`

> |When enabled, MBeans will not be registered for `commons-pool2`.

Reviewer: ```suggestion
|When enabled, MBeans isn't registered for `commons-pool2`.
```

*Issue ID*

* W-11529823
<.^|`com.mulesoft.dw.xml_reader.honourMixedContentStructure`

> |When this property is set to true, DataWeave retains a mixed-content structure instead of grouping text with mixed content into a single text field.

Reviewer 2: To be consistent with the structure, I'd say:
"When `true`, DataWeave..."

> |When this property is set to true, DataWeave retains a mixed-content structure instead of grouping text with mixed content into a single text field.

Reviewer: ```suggestion
|When true, DataWeave retains a mixed-content structure instead of grouping text with mixed content into a single text field.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* MULE-19987
<.^|`enforce.dw.expression.validation`
|When enabled, expression validations are enforced for all DataWeave expressions.

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* MULE-19967
<.^|`force.runtime.profiling.consumers.enablement`
|When enabled, profiling consumers implemented by Mule are enabled by default.

*Available Since*

* 4.5.0

* 4.4.0-202202

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* MULE-19967
<.^|`mule.disable.registryBootstrapOptionalEntries`
|When enabled, `optional` attribute in entries in a `registry-bootstrap.properties` are ignored.

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-10736301
<.^|`mule.disable.applyObjectProcessor`
|When enabled, `org.mule.runtime.core.privileged.registry.ObjectProcessor` implementations aren't applied on objects registered into the `SimpleRegistry`.

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* MULE-11737
<.^|`mule.deployment.validateAppModelWithRegionClassloader`

> |When enabled, the application model is validated with the region classloader. When disabled, it is validated with the application classloader.

Reviewer 2: To avoid the passive voice and a distant reference for the "it", I'd say:
When enabled, the region classloader validates the application model. When disabled, the application classloader validates the application model.

> |When enabled, the application model is validated with the region classloader. When disabled, it is validated with the application classloader.

Reviewer: ```suggestion
|When enabled, the region classloader validates the application model. When disabled, the application classloader validates the application model.
```

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-12207110
<.^|`mule.honour.insecure.tls.configuration`
|When enabled, the insecure TLS configuration is honored even if there are fields of the TrustStore configured.

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-10822938
<.^|`mule.transaction.sink.index`
|When enabled, flux sinks are cached using index as part of the key. If a sink is already in use, a new sink is created.

*Available Since*

* 4.4.0

*Enabled by Default Since*

* 4.0.0

* 4.1.0

* 4.2.0

* 4.3.0

* 4.4.0

* 4.5.0

*Issue ID*

* W-12128703
<.^|`mule.put.trace.id.and.span.id.in.mdc`
|When enabled, the trace ID and span ID are added to the MDC when available.

*Available Since*

* 4.5.0

*Enabled by Default Since*

* 4.5.0

*Issue ID*

* W-12979787
<.^|`mule.enable.mule.specific.tracing.information`
|When enabled, the `ancestor-mule-span-id` value is added to the trace state when a span is propagated.

*Available Since*

* 4.5.0

*Enabled by Default Since*

* Not enabled by default in any Mule version

*Issue ID*

* W-13215870
<.^|`mule.enable.policy.context.parallel.scopes`

> |When enabled, a new (Source) Policy Context is created for the execution of parallel scopes: ParallelForeach, ScatterGather, and Async.

Reviewer 2: Are "ParallelForeach", "ScatterGather", and "Async" parameters or words that should be written in backticks?

Reviewer: They are components. We don't write those with backticks
